subsys/net/l2/ieee802154/ieee802154.c

```diff
@@ -251,22 +251,13 @@ static inline void swap_and_set_pkt_ll_addr(struct net_linkaddr *addr, bool comp
 	switch (mode) {
 	case IEEE802154_ADDR_MODE_EXTENDED:
 		addr->len = IEEE802154_EXT_ADDR_LENGTH;
-
-		if (comp) {
-			addr->addr = ll->comp.addr.ext_addr;
-		} else {
-			addr->addr = ll->plain.addr.ext_addr;
-		}
+		addr->addr = has_pan_id ? ll->plain.addr.ext_addr : ll->comp.addr.ext_addr;
```

Comment:
I see here you used a conditional assignment. This is rarely used in Zephyr. I could not find anything on that in the coding guidelines. Maybe stay with the original if else construction.

Reply:
I don't think there's any requirement regarding conditional assignments in the Coding Guidelines, and it's not really that rarely used in Zephyr codebase. IMO it's fine.

Reply:
@ssharks ok for you if I consider this accepted then?